### PR TITLE
Add some extra config options

### PR DESCRIPTION
**Pull request by casualjim**
adds options that I have in use in a go version of the current library I'm working on: 

* set_pipelined_write,
* memtable_huge_page_size 
* optimize_universal_style_compaction

### CODE REVIEW

File: src/db_options.rs [hunk 1]
```diff
@@ -192,6 +192,12 @@ impl BlockBasedOptions {
         }
     }
 
+    pub fn set_metadata_block_size(&mut self, size: usize) {
```
Comment (aleksuss) [inline src/db_options.rs]: Could you add some description for the public method ?

File: src/db_options.rs [hunk 2]
```diff
@@ -201,6 +207,15 @@ impl BlockBasedOptions {
         }
     }
 
+    pub fn set_lru_cache_compressed(&mut self, size: size_t) {
```
Comment (aleksuss) [inline src/db_options.rs]: Here as well.

**Comment by casualjim** [comment]
I took care of this now

File: src/db_options.rs [hunk 3]
```diff
@@ -186,12 +186,41 @@ impl Drop for ReadOptions {
 }
 
 impl BlockBasedOptions {
+    /// Approximate size of user data packed per block.  Note that the
```
Comment (aleksuss) [inline src/db_options.rs]: Sorry for nagging but decrease amount of spaces between sentences please.

File: src/db_options.rs [hunk 4]
```diff
@@ -201,6 +231,19 @@ impl BlockBasedOptions {
         }
     }
 
+    /// When configured: use the specified cache for compressed blocks.
+    /// Otherwise rocksdb will not use a compressed block cache.
+    /// Note: though it looks similar to `block_cache`, RocksDB doesn't put the
+    ///       same type of object there.
```
Comment (aleksuss) [inline src/db_options.rs]:
```suggestion
    /// same type of object there.
```

File: src/db_options.rs [hunk 5]
```diff
@@ -186,12 +186,42 @@ impl Drop for ReadOptions {
 }
 
 impl BlockBasedOptions {
+    /// Approximate size of user data packed per block.  Note that the
```
Comment (aleksuss) [inline src/db_options.rs]:
```suggestion
    /// Approximate size of user data packed per block. Note that the
```

**Pull request merged**